Automattic/wp-calypso
# Full Site Editing: remove obsolete wp_template CPT

#### Changes proposed in this Pull Request

With the new themes approach we don't really use `wp_template` CPT, and unnecessarily complicates a lot of our code. I'm also removing some functionality that was tied to it like sidebar selector and separate handling in blocks for this page. All of this will be unreachable once we hide the template CPTs.

This separation made sense when we were going for the Blank theme approach, or full page hijacking in the plugin. Now this layout is defined by theme template files (`header.php`, `footer.php`) so having this stored in the template becomes kind of pointless since it can't be manipulated - even if we do it, it wouldn't be reflected on the page.

Next step would be to just rename `wp_template_part` to `wp_template`.

The purpose of this is also to provide better alignment down the road (ping me in Slack if you need clarification on this).

Fixes https://github.com/Automattic/wp-calypso/issues/35009

#### Testing instructions

1. Apply https://github.com/Automattic/themes/pull/1169 to your themes repo.
2. Delete previous Templates and Template Parts entries.
3. Apply this branch to Calypso.
4. Map this theme to your local Gutenberg Docker dev environment, in a similar fashion you do for FSE plugin: `{themes_repo_path}/themes/modern-business:/var/www/html/wp-content/themes/modern-business`
5. Delete `modern-business-fse-template-data` option if it was set on your site. Otherwise this would prevent data population here https://github.com/Automattic/themes/pull/1167/files#diff-8573df061382abec8228a31b5eb02880R84.
6. Activate Modern Business theme to trigger the activation hook.
7. Make sure that Template Parts are populated with correct content.
8. Make sure that everything still works as before.
